### PR TITLE
#2593 pcl install failed, adding qt5 support

### PR DESCRIPTION
**Pull request by ahundt**
https://github.com/Homebrew/homebrew-science/issues/2593

### CODE REVIEW

**Comment by tdsmith** [comment]
The bot choked on the first commit; kicked off a rebuild just now.

**Comment by ahundt** [comment]
yeah I had squashed it down to one commit for simplicity, guess it didn't handle that well?

File: pcl.rb [hunk 1]
```diff
@@ -78,10 +80,15 @@ class Pcl < Formula
   depends_on "qhull"
   depends_on "libusb"
   depends_on "qt" => :recommended
-  if build.with? "qt"
+  depends_on "qt5" => :optional
```
Comment (dpo) [inline pcl.rb]: This is redundant with line 69. Similarly, line 70 should just be `depends_on "qt" => :recommended`.

Comment (ahundt) [inline pcl.rb]: Sorry but I don't totally understand what to do here per your suggestion. I don't know ruby very well so I just modified what's here in a way I knew would work. Could you elaborate on how to make your suggested improvement?

Comment (dpo) [inline pcl.rb]: Sorry, I'm saying that `option "with-qt5", "Build with Qt5"` and `depends_on "qt5" => :optional` are redundant. I'd rather only keep the latter. Similarly, `option "without-qt", "Build without Qt4"` and `depends_on "qt" => :recommended` are also redundant.

**Comment by ahundt** [comment]
I believe I have made the change suggested by @dpo. Is anything else required to merge?

**Comment by ahundt** [comment]
@dpo I apologize but it seems I misunderstood the changes you advised and broke the build when attempting to make the change you suggested. Could you perhaps give me a patch to apply or explain again?

**Comment by dpo** [comment]
Your changes look fine (though the `depends_on` should have stayed where they were). It seems there was a bot issue. I'll kick off a new build.

**Comment by ahundt** [comment]
hmm I'm having a tough time getting back to a version that works

**Comment by dpo** [comment]
Here's the error
```
-- Installing: /usr/local/Cellar/cmake/3.3.1/share/cmake/Modules/FindCMinpack.cmake
CMake Error at cmake/cmake_install.cmake:43 (file):
  file INSTALL cannot copy file
  "/tmp/cminpack20150827-88183-1dt9mst/cminpack-1.3.4/cmake/FindCMinpack.cmake"
  to "/usr/local/Cellar/cmake/3.3.1/share/cmake/Modules/FindCMinpack.cmake".
Call Stack (most recent call first):
  cmake_install.cmake:47 (include)

make: *** [install] Error 1
==> Sandbox profile:
(version 1)
(debug deny) ; log all denied operations to /var/log/system.log
(allow file-write* (subpath "/private/tmp"))
(allow file-write* (subpath "/private/var/tmp"))
(allow file-write* (regex #"^/private/var/folders/[^/]+/[^/]+/[C,T]/"))
(allow file-write* (subpath "/private/tmp"))
(allow file-write* (subpath "/Library/Caches/Homebrew"))
(allow file-write* (subpath "/Users/brew/Jenkins/science/version/mavericks/logs/cminpack"))
(allow file-write* (subpath "/Users/brew/Library/Developer/Xcode/DerivedData"))
(allow file-write* (subpath "/usr/local/Cellar/cminpack"))
(allow file-write* (subpath "/usr/local/etc"))
(allow file-write* (subpath "/usr/local/var"))
(allow file-write*
    (literal "/dev/ptmx")
    (literal "/dev/dtracehelper")
    (literal "/dev/null")
    (regex #"^/dev/fd/[0-9]+$")
    (regex #"^/dev/ttys?[0-9]*$")
    )
(deny file-write*) ; deny non-whitelist file write operations
(allow default) ; allow everything else
```
I'm afraid I can't help much. `pcl` doesn't even begin to build on my machine.

**Comment by ahundt** [comment]
I've identified the problem, there needs to be a way to make the following directory writable:
`/usr/local/Cellar/cmake/3.3.1/share/cmake/Modules/`

It isn't in the list of `allow file-write*` in the comment you posted. Should I create a separate issue to ask about this?

**Comment by tdsmith** [comment]
@BrewTestBot test this please

**Comment by ahundt** [comment]
@xu-cheng seemed to explain what to do in https://github.com/Homebrew/homebrew-science/issues/2696, could I possibly get assistance with pushing this through so the issue can be merged?

**Comment by ahundt** [comment]
Ok it seems since #2701 was fixed, @tdsmith tried re-running this to see if it passed since no files from a dependency would be installed in the wrong place. Unfortunately there is still a problem that remains and needs to be investigated/resolved.

**Comment by ahundt** [comment]
hmm seems to still be failing on cminpack if I'm reading the build errors correctly, should #2701 be reopened?

**Comment by dunn** [comment]
Just an audit failure as far as I can tell:
```
        Error: 13 problems in 1 formula
homebrew/science/pcl:
 * Use spaces instead of tabs for indentation
 * Use spaces instead of tabs for indentation
 * Use spaces instead of tabs for indentation
 * Use spaces instead of tabs for indentation
 * Use spaces instead of tabs for indentation
 * Use spaces instead of tabs for indentation
 * Use spaces instead of tabs for indentation
 * Use spaces instead of tabs for indentation
 * Use spaces instead of tabs for indentation
 * Use spaces instead of tabs for indentation
 * Use spaces instead of tabs for indentation
 * Use spaces instead of tabs for indentation
 * Use spaces instead of tabs for indentation

File: pcl.rb [hunk 2]
```diff
@@ -160,8 +179,8 @@ def install
 +++ b/cmake/pcl_find_cuda.cmake
 @@ -1,16 +1,6 @@
  # Find CUDA
- 
- 
+
```
Comment (dunn) [inline pcl.rb]: Keep the whitespace in the patch, removing it can sometimes prevent it from applying.

Comment (ahundt) [inline pcl.rb]: oh whoops removed those tabs too, well let's see if it applies correctly in the build. I can put the tabs from the patch back if needed

Comment (dunn) [inline pcl.rb]: Oh, if some of the tabs are in the patch then leave those, yeah!

Comment (ahundt) [inline pcl.rb]: seems to have worked!

File: pcl.rb [hunk 3]
```diff
@@ -173,16 +192,16 @@ def install
 -endif()
 -
  if(MSVC11)
- 	# Setting this to true brakes Visual Studio builds.
- 	set(CUDA_ATTACH_VS_BUILD_RULE_TO_CUDA_FILE OFF CACHE BOOL "CUDA_ATTACH_VS_BUILD_RULE_TO_CUDA_FILE")
+   # Setting this to true brakes Visual Studio builds.
```
Comment (dunn) [inline pcl.rb]: Sorry, I meant leave the inline patch as-is.

Comment (ahundt) [inline pcl.rb]: yeah I knew that but I had already committed the change, seems to work anyway so probably good

**Comment by ahundt** [comment]
What's left to merge this? Thanks!

File: pcl.rb (hunk 3 above)
Comment (dunn) [inline pcl.rb]: The patch is only for HEAD builds, so the bot won't use it.  Can you try building HEAD locally and see if the patch applies?

Comment (ahundt) [inline pcl.rb]: It seems that even if I revert back to the tabs in the patch it doesn't apply. I reverted the specific lines and ran the following:

```
± brew reinstall pcl --with-qt5 --with-openni2 --with-examples
==> Reinstalling homebrew/science/pcl with --with-qt5, --with-openni2, --with-examples
==> Cloning https://github.com/PointCloudLibrary/pcl.git
Updating /Library/Caches/Homebrew/pcl--git
==> Checking out branch master
==> Patching
patching file cmake/pcl_find_cuda.cmake
Hunk #2 FAILED at 37.
1 out of 2 hunks FAILED -- saving rejects to file cmake/pcl_find_cuda.cmake.rej
Error: Failure while executing: /usr/bin/patch -g 0 -f -p1
```

**Comment by indianajohn-aemass** [comment]
I'm just popping in to request that this patch gets merged. A bug in Qt4 that will never be fixed ( https://bugreports.qt.io/browse/QTBUG-40415 ) has forced me to use QT5 with a PCL application. This necessitates that I build PCL with QT5, and at the moment I'm maintaining my own patches. This request being merged would have made my life a lot easier.

File: pcl.rb (hunk 1 above)
Comment (dpo) [inline pcl.rb]: The above options are still redundant.

**Comment by ahundt** [comment]
I'm kinda stuck myself on this, I think an actual maintainer of homebrew-science is needed to finish this up.

File: pcl.rb (hunk 1 above)
Comment (dunn) [inline pcl.rb]: @ahundt did you address this?

**Comment by ahundt** [comment]
does 4d8bf6f mean this pull request should be closed?

**Pull request closed without merging**